kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels/ManageSelectedLessonResource.vue

```diff
@@ -0,0 +1,207 @@
+          </h1>
+        </div>
+      </template>
+
+      <SelectedResources
+        :resourceList="resources"
+        :currentLesson="currentLesson"
+        :loading="resources.length === 0"
+        @removeResource="removeResource"
+        @navigateToParent="navigateToParent"
+      />
+
+      <template #bottomNavigation>
+        <div class="bottom-buttons-style">
+          <KButton
+            :primary="true"
+            :text="saveLessonResources$()"
+            @click="closeSidePanel()"
+          />
+        </div>
+      </template>
+    </SidePanelModal>
+
+    <KModal
+      v-if="showModal"
+      :submitText="coreString('continueAction')"
+      :cancelText="coreString('cancelAction')"
+      :title="closeConfirmationTitle$()"
+      @cancel="closeModal"
+      @submit="$router.go(-1)"
+    >
+      {{ closeConfirmationMessage$() }}
+    </KModal>
+  </div>
+
+</template>
+
+
+<script>
+
+  import SidePanelModal from 'kolibri-common/components/SidePanelModal';
+  import { mapState } from 'vuex';
+  import { searchAndFilterStrings } from 'kolibri-common/strings/searchAndFilterStrings';
+  import commonCoreStrings from 'kolibri/uiText/commonCoreStrings';
+  import { ref, getCurrentInstance } from 'vue';
+  import { enhancedQuizManagementStrings } from 'kolibri-common/strings/enhancedQuizManagementStrings';
+  import { PageNames } from '../../../../constants';
+  import commonCoach from '../../../common';
+  import SelectedResources from './SelectedResources';
+
+  export default {
+    name: 'ManageSelectedLessonResources',
+    components: {
+      SidePanelModal,
+      SelectedResources,
+    },
+    mixins: [commonCoach, commonCoreStrings],
+    setup() {
+      const showModal = ref(false);
+      const hasArrayChanged = ref(false);
+      const router = getCurrentInstance().proxy.$router;
+
+      const { saveLessonResources$, numberOfSelectedResource$ } = searchAndFilterStrings;
+      const { closeConfirmationTitle$, closeConfirmationMessage$ } = enhancedQuizManagementStrings;
+
+      function closeModal() {
+        showModal.value = false;
+      }
+
+      function hasModifiedArray() {
+        hasArrayChanged.value = true;
+      }
+
+      function closeSidePanel() {
+        if (hasArrayChanged.value) {
+          showModal.value = true;
+        } else {
+          router.go(-1);
+        }
+      }
+
+      return {
+        saveLessonResources$,
+        numberOfSelectedResource$,
+        closeConfirmationTitle$,
+        closeConfirmationMessage$,
+        showModal,
+        closeModal,
+        closeSidePanel,
+        hasModifiedArray,
+      };
+    },
+    data() {
+      return {
+        resources: [],
+      };
+    },
+    computed: {
+      ...mapState('lessonSummary', ['currentLesson', 'workingResources', 'resourceCache']),
+    },
+    mounted() {
+      setTimeout(() => {
+        this.getResources();
+      }, 2000);
+    },
```

Comment:
This will be taken off- was for testing 
The reason for the delay was to enable the component load fully with data, otherwise the function was returning undefined.